openmls/src/group/core_group/validation.rs

```diff
@@ -208,10 +244,14 @@ impl CoreGroup {
                 .remove_proposals()
                 .any(|p| p.remove_proposal().removed == index);
             // ValSem104
+            // Note: This is different to the leaf node check above. It checks for duplicate
```

Comment:
For both of these: Do we need to check twice or would it be enough to check once, after the proposals are applied?